core/client/src/main/java/alluxio/client/lineage/LineageFileSystem.java

```diff
@@ -41,7 +42,7 @@ public static synchronized LineageFileSystem get() {
   }
 
   protected LineageFileSystem() {
```

Comment:
Injecting here is pretty simple so I've pushed the dependency back to the static getter. If we go further the world needs to start to change :)